src/Compilers/CSharp/Test/Semantic/IOperation/IOperationTests_INoPiaObjectCreationOperation.cs

```diff
@@ -0,0 +1,525 @@
+            string pia = @"
+using System;
+using System.Runtime.InteropServices;
+using System.Runtime.CompilerServices;
+
+[assembly: PrimaryInteropAssemblyAttribute(1,1)]
+[assembly: Guid(""f9c2d51d-4f44-45f0-9eda-c9d599b58257"")]
+
+[ComImport()]
+[Guid(""f9c2d51d-4f44-45f0-9eda-c9d599b58279"")]
+[CoClass(typeof(ClassITest33))]
+public interface ITest33
+{
+}
+
+[Guid(""f9c2d51d-4f44-45f0-9eda-c9d599b58278"")]
+public abstract class ClassITest33
+{
+}
+";
+
+            var piaCompilation = CreateCompilation(pia, options: TestOptions.ReleaseDll);
+
+            CompileAndVerify(piaCompilation);
+
+            string consumer = @"
+class UsePia
+{
+    public void M1(ITest33 x, int y)
+    {
+	    x = /*<bind>*/new ITest33()/*</bind>*/;
+    }
+} 
+";
+
+            string expectedOperationTree = @"
+INoPiaObjectCreationOperation (OperationKind.None, Type: ITest33) (Syntax: 'new ITest33()')
+  Initializer: 
+    null
+";
+            var expectedDiagnostics = DiagnosticDescription.None;
+
+            VerifyOperationTreeAndDiagnosticsForTest<ObjectCreationExpressionSyntax>(consumer, expectedOperationTree, expectedDiagnostics, references: new[] { piaCompilation.EmitToImageReference(embedInteropTypes: true) });
+        }
+
+        [CompilerTrait(CompilerFeature.IOperation, CompilerFeature.Dataflow)]
+        [Fact]
+        public void NoPiaObjectCreationFlow_01()
+        {
+            string pia = @"
+using System;
+using System.Runtime.InteropServices;
+using System.Runtime.CompilerServices;
+
+[assembly: PrimaryInteropAssemblyAttribute(1,1)]
+[assembly: Guid(""f9c2d51d-4f44-45f0-9eda-c9d599b58257"")]
+
+[ComImport()]
+[Guid(""f9c2d51d-4f44-45f0-9eda-c9d599b58279"")]
+[CoClass(typeof(ClassITest33))]
+public interface ITest33 : System.Collections.IEnumerable
+{
+    void Add(int x);
+}
+
+[Guid(""f9c2d51d-4f44-45f0-9eda-c9d599b58278"")]
+public abstract class ClassITest33
+{
+}
+";
+
+            var piaCompilation = CreateCompilation(pia, options: TestOptions.ReleaseDll);
+
+            CompileAndVerify(piaCompilation);
+
+            string consumer = @"
+class UsePia
+{
+    /*<bind>*/public void M1(ITest33 x, int y)
+    {
+	    x = new ITest33  { y };
+    }/*</bind>*/
+} 
+";
+
+            var expectedDiagnostics = DiagnosticDescription.None;
+
+            string expectedFlowGraph = @"
+Block[B0] - Entry
+    Statements (0)
+    Next (Regular) Block[B1]
+Block[B1] - Block
+    Predecessors: [B0]
+    Statements (4)
+        IFlowCaptureOperation: 0 (OperationKind.FlowCapture, Type: null, IsImplicit) (Syntax: 'x')
+          Value: 
+            IParameterReferenceOperation: x (OperationKind.ParameterReference, Type: ITest33) (Syntax: 'x')
+
+        IFlowCaptureOperation: 1 (OperationKind.FlowCapture, Type: null, IsImplicit) (Syntax: 'new ITest33  { y }')
+          Value: 
+            INoPiaObjectCreationOperation (OperationKind.None, Type: ITest33) (Syntax: 'new ITest33  { y }')
+              Initializer: 
+                null
```

Comment:
Why is the initializer null here? I would expect it to have the collection initializer value.

Reply:
> Why is the initializer null here? I would expect it to have the collection initializer value. 

Object creation is broken into individual operations: an actual instance creation followed by individual initializations.

---
In reply to: [193145223](https://github.com/dotnet/roslyn/pull/27476#discussion_r193145223) [](ancestors = 193145223)